src/main/java/teammates/ui/automated/InstructorCourseJoinEmailWorkerAction.java

```diff
@@ -39,7 +39,10 @@ public void execute() {
         CourseAttributes course = logic.getCourse(courseId);
         Assumption.assertNotNull(course);
 
-        InstructorAttributes instructor = logic.getInstructorForEmail(courseId, instructorEmail);
+        // The instructor is queried using the `id` (email%courseId) as it ensures that the
+        // instructor is retrieved (and not null) even if the index building for
+        // saving the new instructor takes more time in GAE.
+        InstructorAttributes instructor = logic.getInstructorById(courseId, instructorEmail);
```

Comment:
I actually want to explicitly say that instructor ID can be constructed back by courseID + instructorEmail here (because instructors' email cannot be changed before joining the course).

Reply:
Updated. Also corrected the typo in `Instructor` entity